src/sentry/api/serializers/models/event.py

```diff
@@ -221,3 +221,45 @@ def serialize(self, obj, attrs, user):
         result['entries'] = [e for e in result['entries']
                              if e['type'] != 'breadcrumbs']
         return result
+
+
+class SnubaEvent(object):
+    """
+        A simple wrapper class on a row (dict) returned from snuba representing
+        an event. Provides a class name to register a serializer against, and
+        Makes keys accessible as attributes.
+    """
+
+    # The list of columns that we should request from snuba to be able to fill
+    # out a proper event object.
+    selected_columns = [
+        'event_id',
+        'message',
+        'user_id',
+        'username',
+        'email',
+    ]
+
+    def __init__(self, kv):
+        assert set(kv.keys()) == set(self.selected_columns)
+        self.__dict__ = kv
```

Comment:
Could a namedtuple work here? You would be able to unpack the dict into the named tuple and get errors if there were extra/not enough keys. :bike: :house:

Reply:
`namedtuple` and its subclasses will try and [serialize as a JSON array even if a custom serializer is registered](https://github.com/getsentry/sentry/blob/8941b52edf4eca86adfaef8d20e3ede65f0163b0/src/sentry/api/serializers/base.py#L8-L42), so no, not without changes there.

Reply:
Didn't know that. Thank you.

Reply:
I only know this because we've spent a few hours of our life trying to figure out why that didn't work in the past. 😄